### PR TITLE
[B+C] Add breakBlock param to createExplosion - Fixes BUKKIT-5350

### PR DESCRIPTION
**Pull request by stuntguy3000**
**The Issue:**
Plugins should be able to create explosions, with locations, and specify if blocks should be modified.

**Justification for this PR:**
There is currently an API for creating explosions with no block damage, however that only allows x y z co-ordinates. It would be nice to be able to pass a Location and have this parameter.

**JIRA Ticket:**
https://bukkit.atlassian.net/browse/BUKKIT-5350

**Testing Material**

```
    public void onEnable() {
        getCommand("test").setExecutor(this);
    }

    public boolean onCommand(CommandSender s, Command cmd, String label, String[] args) {
        if (s instanceof Player) {
            Player p = (Player) s;

            p.getWorld().createExplosion(p.getLocation(), 1f, false, false);
                        // When executed, a explosion with no block damage is summoned.
        }
        return true;
    }
```

Associated CraftBukkit Ticket: https://github.com/Bukkit/CraftBukkit/pull/1328
Associated Bukkit Ticket: https://github.com/Bukkit/Bukkit/pull/1017

### CODE REVIEW

**Comment by stuntguy3000** [comment]
Just compiled Bukkit and CraftBukkit and the changes work.

File: src/main/java/org/bukkit/World.java [hunk 1]
```diff
@@ -627,6 +627,18 @@
     public boolean createExplosion(Location loc, float power, boolean setFire);
 
     /**
+     * Creates explosion at given coordinates with given power and optionally
```
Comment (turt2live) [inline src/main/java/org/bukkit/World.java:630]: English

**Comment by turt2live** [comment]
CONTRIBUTING.md needs to be followed.

You have not provided testing materials and you link to a closed PR.

**Comment by stuntguy3000** [comment]
1) Fixing above
2) I didn't know testing materials had to be actually followed
3) I created another pr, but it was closed as a dupe and was told to edit original pr..

**Comment by Wolvereness** [comment]
Associated [ticket](http://cbukk.it/j/BUKKIT-5350) closed.

**Pull request closed without merging**